packages/coil-extension/src/content/services/ContentScript.ts

```diff
@@ -44,46 +57,53 @@ export class ContentScript {
   ) {}
 
   handleMonetizationTag() {
-    const { runtime, monetization, window } = this
-
-    function startMonetization(details: PaymentDetails) {
-      const request: StartWebMonetization = {
-        command: 'startWebMonetization',
-        data: { ...details, initiatingUrl: window.location.href }
-      }
-
-      monetization.setMonetizationRequest({
+    const startMonetization = async (details: PaymentDetails) => {
+      this.monetization.setMonetizationRequest({
         paymentPointer: details.paymentPointer,
-        requestId: details.requestId
+        requestId: details.requestId,
+        initiatingUrl: details.initiatingUrl
       })
-      runtime.sendMessage(request)
+      if (this.frames.isIFrame) {
+        const allowed = !this.frames.isDirectChildFrame
+          ? false
+          : await new Promise<boolean>(resolve => {
+              const message: CheckIFrameIsAllowedFromIFrameContentScript = {
+                command: 'checkIFrameIsAllowedFromIFrameContentScript'
+              }
+              this.runtime.sendMessage(message, resolve)
+            })
+        if (!allowed) {
+          console.error(
```

Comment:
It might be nicer to log the actual element from the parent frame content script